jupyter/analysis-automation/BasicExample.ipynb

```diff
@@ -0,0 +1,238 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import asyncio\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "from systemlink.clients.nitdmreader.api.metadata_api import MetadataApi\n",
+    "from systemlink.clients.nitdmreader.models.file_list import FileList\n",
+    "import systemlink.clients.nitdmreader as tdmreader\n",
+    "\n",
+    "metadata_api = MetadataApi()\n",
+    "data_api = tdmreader.DataApi()\n",
+    "#file_list = tdmreader.FileList()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Metadata helper method \n",
+    "\n",
+    "def get_property(element, property_name):\n",
+    "    return next((e.value for e in element.properties.properties if e.name == property_name), None)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use data links to read data\n",
+    "data_links = ni_analysis_automation[\"data_links\"]\n",
+    "file_ids = [d[\"fileId\"] for d in data_links]\n",
+    "file_infos = await metadata_api.get_multiple_file_info(FileList(file_ids))\n",
+    "file_info = file_infos[0]\n",
+    "\n",
+    "test_file_name = get_property(file_info, \"name\")\n",
+    "program_name = get_property(file_info, \"Test~Procedure\") or get_property(file_info, file_name_prop)\n",
+    "\n",
+    "group_names = []\n",
+    "channels = []\n",
+    "formatted_properties = []\n",
+    "\n",
+    "for group in file_info.groups:\n",
+    "    group_names.append(group.name)\n",
+    "    channels.append(group.channels)\n",
+    "    max_values_of_group = []    \n",
+    "    min_values_of_group = []    \n",
+    "    mean_values_of_group = []    \n",
+    "\n",
+    "    for channel in group.channels:\n",
+    "        minimum = float(get_property(channel, \"minimum\") or \"NaN\")\n",
+    "        maximum = float(get_property(channel, \"maximum\") or \"NaN\")\n",
+    "        mean_values_of_group.append((minimum + maximum) / 2)\n",
+    "        max_values_of_group.append(maximum)\n",
+    "        min_values_of_group.append(minimum)\n",
+    "\n",
+    "        \n",
+    "    # Calculate statistic values from metadata \n",
+    "    abs_min = max(max_values_of_group)\n",
+    "    abs_max = min(max_values_of_group)\n",
+    "    abs_mean = sum(mean_values_of_group) / float(len(mean_values_of_group))\n",
+    "    formatted_properties.append(\"Absolute Maximum: \"+str(round(abs_max,3))+\" °C\"+\n",
+    "                                \",Absolute Minimum: \"+str(round(abs_min,3))+\" °C\"+\n",
+    "                                \",Mean Value: \"+str(round(abs_mean,3))+\" °C\") "
```

Comment:
I would prefer using Pythons string interpolation with f-Strings. E.g. you can combine a string and a number while using only 3 decimal digits with

```python
f"Absolute Maximum: {abs_max:.3f} °C"
```

Reply:
Good idea, thank you